core/src/main/scala/kafka/log/Log.scala

```diff
@@ -557,8 +587,8 @@ class Log(@volatile var dir: File,
     var entry = segments.floorEntry(startOffset)
 
     // attempt to read beyond the log end offset is an error
-    if(startOffset > next || entry == null)
-      throw new OffsetOutOfRangeException("Request for offset %d but we only have log segments in the range %d to %d.".format(startOffset, segments.firstKey, next))
+    if(startOffset > next || entry == null || startOffset < logStartOffset)
```

Comment:
Sure. I updated the comment to say `return error on attempt to read beyond the log end offset or read below log start offset`. Is this OK?